tests/rule-engine/forwardchainer/fc-unsatisfied-premise.scm

```diff
@@ -0,0 +1,32 @@
+(define (return-value x) x)
+
+(define rule
+ (BindLink
+
+  (VariableList
+   (TypedVariable (Variable "$a") (Type "ConceptNode"))
+   (TypedVariable (Variable "$b") (Type "ConceptNode"))
+   (TypedVariable (Variable "$c") (Type "ConceptNode"))
+   (TypedVariable (Variable "$d") (Type "ConceptNode")))
+
+  (And
+   (Inheritance
+    (Variable "$a")
+    (Variable "$b"))
+   (Member
+    (Variable "$c")
+    (Variable "$d")))
```

Comment:
@Yidnekachew to test that well you'd need to have all clauses turned into constants. You may for instance replace
```scheme
 (Member
    (Variable "$c")
    (Variable "$d"))
```
by
```scheme
(Member
    (Variable "$a")
    (Variable "$b"))
```
because I suspect this would pass even before your fix (commit 9547797d3).